README.md

```diff
@@ -1,2 +1,61 @@
+
+Eventually you should see the flow finish with "Task finished successfully." Once you see that, you know you've got a working minimal flow: you're now ready to start putting your model training code _into_ this flow to run it on the Outerbounds instance. Comments _inside_ the template file should help you understand where to put that code, and demonstrate some of the tools Metaflow makes available to you.
+
+ ## Next: A Fancy Dashboard
+
+An admin from Mozilla’s MLOps team needs to set you up with a Weights and Biases User account, and also a team if your team doesn’t have that yet, on Weights and Biases. Reach out to #mlops in Slack for help with this.
+
+MLOps needs to know:
+1. If you will need a service account for your Weights and Biases team—say, to set up CI or something. Seats cost significant amounts of money, so if you need extra accounts “just in case,” no you don’t. But if you are, for example, already set up with CI for your model and you want that on Weights and Biases, we can get you a service account for that.
+
+Once you have the account, you're in luck; the `template_flow` already integrates with Weights and Biases for you. 
+
+Note the three environment variables listed at the top of the `train` step:
+
+- The WANDB_API_KEY you can get by going to this page, once your wandb credentials are set up (see step 1). Treat this with the sensitive that you would treat any API key. You will see in a later step how to use this string; for now, you can stick this in your local repository’s .env. 
+
+- The WANDB_ENTITY is your wandb username (or, in the case of a service account, the service account username, which MLOps should have provided to you if you had them make a service account). 
+
+- The WANDB_PROJECT is the name of the project that this run is for. You can see and create projects for your team at this URL: https://wandb.ai/YOUR_TEAM_NAME_HERE/projects.
+
+You can customize what you send to Weights and Biases as well as the graphs and data that appears there, and the team maintains decent documentation on your customization options. The wandb engineering support team is also incredibly helpful. Talk to MLOps about your problem and they can get you added to our joint channel with wandb where folks are available to help.
+
+
+When it’s time to run your flow on outerbounds with your weights and biases dashboard set up, the Outerbounds instance will not have the three environment variables we just talked about. We are currently working on that with them for things like scheduled flows. In the meantime, when you kick off the flow from your local machine, you can specify the environment variables via our super-advanced technical workaround: tacking them onto the front of the command. To wit:
```

Comment:
```suggestion
To run the template flow and test the connection to W&B, specify the environment variables on the CLI:
```

Reply:
I'd like to take this instance as an example of the kind of thing I'd would have liked to say in a conversation about why I write documentation this way for this project.

Engineers are not going to like that this is how they have to specify credentials, and they are right; it is objectively absurd, inconvenient, and furthermore insecure, as it means they'll have their unencrypted `WAND_API_KEY`s repeated ad infinitum in their console history. 

There should be, and will be, a better way to do this. _Acknowledging_ this kind of thing up front reassures our constituents that we know what this user experience is like, and we have plans to change it. That engenders trust in both our technical acumen and our empathy for their experience. Since it is our explicit goal to build close relationships without constituents as purveyors of an internal tool, that _branding_—"Hi! We're a team that's here to help you! We know this part fails a little bit in that mission, and we're working on it!" makes a difference in constituent experience at scale over time.

Reply:
> Engineers are not going to like that this is how they have to specify credentials, and they are right; it is objectively absurd, inconvenient, and furthermore insecure, as it means they'll have their unencrypted `WAND_API_KEY`s repeated ad infinitum in their console history.

This is a problem that we'll need to solve and, once we do, we'll update the docs.
I don't feel strongly about how we pass API keys around in the example here, prepending them in the CLI is one way, but we could document other ways you'd recommend to pass them around instead!

> There should be, and will be, a better way to do this. _Acknowledging_ this kind of thing up front reassures our constituents that we know what this user experience is like, and we have plans to change it. That engenders trust in both our technical acumen and our empathy for their experience. Since it is our explicit goal to build close relationships without constituents as purveyors of an internal tool, that _branding_—"Hi! We're a team that's here to help you! We know this part fails a little bit in that mission, and we're working on it!" makes a difference in constituent experience at scale over time.

I believe the best way to acknowledge this problem is prioritize its resolution and address it. I don't think making this _specific document_ longer would help with building trust or empathy, but I'm happy to take a step back on this one.

Reply:
> I don't feel strongly about how we pass API keys around in the example here, prepending them in the CLI is one way, but we could document other ways you'd recommend to pass them around instead!

I don't disagree with the decision, in context, to prepend the key to the command: it is the option we have in the absence of Outerbounds having provided the alternative yet, which is why you made the (in my opinion pragmatic) decision to do it this way, and why we're recommending it at the moment. The decision to do this is not what's inconvenient and absurd; it's the context _forcing_ us to do it that's inconvenient and absurd. I apologize for not making this clearer. 

> I believe the best way to acknowledge this problem is prioritize its resolution and address it. I don't think making this specific document longer would help with building trust or empathy, but I'm happy to take a step back on this one.

I disagree. First, in this specific case, we don't have control over the resolutio: _our_ implementation isn't the blocker. _Outerbounds's_ is. But because we don't want to sour people on the tool, I agree with you that we shouldn't mention this to constituents. But I think we can still acknowledge "this is bad, we know, we're working on it." 

Second, I've seen a lot of cases where leadership teams—a term I'll use generally to describe any team that has power over a group of constituents, which I think includes library maintainers—lost the confidence of their constituents because, despite the fact that they were working hard to fix something, they didn't tell their constituents that. To constituents, that's indistinguishable from being ignored, which is what they'll often assume without (and sometimes unfortunately even _with_) a pre-established trust relationship. 

There'll be times when, for legal or regulatory or privacy or security reasons, we _have_ to keep our work confidential and just absorb ire from our constituents over the status quo, or distrust that we plan to change it. But I'd like to minimize those occasions, and not do it any more often than we have to.